storage/replica_raftstorage.go

```
@@ -83,6 +83,9 @@ func (r *Replica) InitialState() (raftpb.HardState, raftpb.ConfState, error) {
 // TODO(bdarnell): consider caching for recent entries, if rocksdb's builtin caching
 // is insufficient.
 func (r *Replica) Entries(lo, hi, maxBytes uint64) ([]raftpb.Entry, error) {
+	if lo > hi {
+		return nil, util.Errorf("lo:%d is greater than hi:%d", lo, hi)
```

if you can return an error, why not? the callers will do the panicking.